packages/react/src/hooks/useERC20TokenPrice.ts

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from "react";
+import { useNetwork } from "wagmi";
+import { isAddress, getAddress } from "@ethersproject/address";
+import { ChainId } from "@carrot-kpi/sdk";
+
+const DEFILLAMA_API_CHAIN_PREFIX: Record<ChainId, string> = {
+    [ChainId.GNOSIS]: "xdai",
+    [ChainId.SEPOLIA]: "",
+};
+
+const FRESHNESS_THRESHOLD = "1h";
+const CONFIDENCE_THRESHOLD = 0.8;
+
+export const useERC20TokenPrice = (
+    tokenAddress?: string
+): { loading: boolean; price: number } => {
+    const { chain } = useNetwork();
+
+    const [loading, setLoading] = useState(false);
+    const [price, setPrice] = useState(0);
+
+    useEffect(() => {
+        let cancelled = false;
+        const fetchPrice = async () => {
+            if (!tokenAddress || !isAddress(tokenAddress) || !chain) return;
+            const apiChainPrefix =
+                DEFILLAMA_API_CHAIN_PREFIX[chain.id as ChainId];
+            if (!apiChainPrefix) return;
+            if (!cancelled) setLoading(true);
+            try {
+                const checksummedAddress = getAddress(tokenAddress);
+                const keys = [`coingecko:${checksummedAddress}`];
+                if (apiChainPrefix)
```

Comment:
I think we can remove this check here right? We are already checking for its existence before in:
```ts
if (!apiChainPrefix) return;
```
Here it would always exist.

Reply:
Good catch, but we actually need to remove the first check. Will do that.

Reply:
Addressed in 78915bacf99b4945f1fa07e5ec22fdb18f8aca60